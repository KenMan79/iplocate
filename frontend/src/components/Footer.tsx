import React from "react";
import styled from "styled-components";

export const Footer: React.FC = () => (
  <Wrapper>
    <div>Â© 2020, Jeff Hackshaw</div>
    <MaxMindNotice>
      This product includes GeoLite2 data created by MaxMind, available from{" "}
      <a href="https://www.maxmind.com">https://www.maxmind.com</a>.
    </MaxMindNotice>
  </Wrapper>
);

const Wrapper = styled.footer`
  width: 100%;
  display: flex;
  flex-flow: row wrap;
  justify-content: space-between;
  align-items: center;
  padding: 1rem 2rem;
  background-color: var(--background-card);

  @media screen and (max-width: 450px) {
    flex-flow: column nowrap;
    align-items: flex-start;
    padding: 1rem;
  }
`;

const MaxMindNotice = styled.div`
  color: var(--text-light);
  font-weight: 300;
  font-size: 0.7rem;
  text-align: right;
  padding-top: 1rem;

  @media screen and (max-width: 450px) {
    text-align: left;
  }

  a: {
    color: var(--text-secondary);
  }
`;
