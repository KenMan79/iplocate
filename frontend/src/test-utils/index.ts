export * from "./test-utils";
export * from "./ipdata";
