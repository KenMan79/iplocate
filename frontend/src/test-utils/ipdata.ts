import { IP } from "types";

export const baseIP: IP = {
  continent: {
    code: "EU",
    geonameId: 6255148,
    names: {
      en: "Europe",
    },
  },
  country: {
    geonameId: 3017382,
    isInEuropeanUnion: true,
    isoCode: "FR",
    names: {
      en: "France",
    },
  },
  traits: {
    isAnonymous: false,
    isAnonymousProxy: false,
    isAnonymousVpn: false,
    isHostingProvider: false,
    isLegitimateProxy: false,
    isPublicProxy: false,
    isResidentialProxy: false,
    isSatelliteProvider: false,
    isTorExitNode: false,
    ipAddress: "2.2.2.2",
    network: "2.2.0.0/18",
  },
  city: {},
  location: {
    accuracyRadius: 500,
    latitude: 48.8582,
    longitude: 2.3387,
    timeZone: "Europe/Paris",
  },
};
