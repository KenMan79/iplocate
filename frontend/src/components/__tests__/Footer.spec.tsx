import React from "react";
import { Footer } from "../Footer";
import { render } from "test-utils/test-utils";

it("matches snapshot", () => {
  const { asFragment } = render(<Footer />);
  expect(asFragment()).toMatchSnapshot();
});
