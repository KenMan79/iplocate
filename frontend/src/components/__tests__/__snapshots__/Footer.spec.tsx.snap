// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1rem 2rem;
  background-color: var(--background-card);
}

.c1 {
  color: var(--text-light);
  font-weight: 300;
  font-size: 0.7rem;
  text-align: right;
}

.c1 a: {
  color: var(--text-secondary);
}

@media screen and (max-width:450px) {
  .c0 {
    -webkit-flex-flow: column nowrap;
    -ms-flex-flow: column nowrap;
    flex-flow: column nowrap;
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
    padding: 1rem;
  }
}

@media screen and (max-width:450px) {
  .c1 {
    padding-top: 0.5rem;
    text-align: left;
  }
}

<footer
    class="c0"
  >
    <div>
      Â© 2020, Jeff Hackshaw
    </div>
    <div
      class="c1"
    >
      This product includes GeoLite2 data created by MaxMind, available from 
      <a
        href="https://www.maxmind.com"
      >
        https://www.maxmind.com
      </a>
      .
    </div>
  </footer>
</DocumentFragment>
`;
